code/game/objects/items/robot/robot_upgrades.dm

```diff
@@ -401,3 +401,17 @@
 
 	R.make_shell(src)
 	return TRUE
+
+/obj/item/borg/upgrade/traitorsec
+	name = "Oppressor upgrade module"
+	desc = "Illegal upgrade board that unlocks the banned security module."
+	icon_state = "cyborg_upgrade3"
+	origin_tech = "data=4;combat=4;syndicate=3"
+
+/obj/item/borg/upgrade/traitorsec/action(mob/living/silicon/robot/R,mob/living/user)
+	if(..())
+		return FALSE

```

Comment:
what is the purpose of this line

Reply:
I'm honestly not sure, Anturk told me to do it.

Reply:
The parent of `action` contains the insertion checks (prevents use on dead borgs and enforces module restrictions on upgrades designed for a specific module). You can safely remove the `FALSE`, but you must keep the return. The attackby in `robot.dm` calls this to check if the upgrade was successfully installed.